src/Tasks/Unzip.cs

```diff
@@ -131,6 +131,14 @@ private void Extract(ZipArchive sourceArchive, DirectoryInfo destinationDirector
             {
                 FileInfo destinationPath = new FileInfo(Path.Combine(destinationDirectory.FullName, zipArchiveEntry.FullName));
 
+                // Zip archives can have directory entries listed explicitly; we should skip these.
+                // The parent directories of any file entries is created just before file creation.
+                char lastChar = destinationPath.FullName[destinationPath.FullName.Length - 1];
+                if (lastChar == Path.DirectorySeparatorChar || lastChar == Path.AltDirectorySeparatorChar)
```

Comment:
IIRC, on Linux and macOS, both `DirectorySeparatorChar` and `AltDirectorySeparatorChar` are `/`. If the zip format allows `\`, we should probably explicitly check for both here.

Reply:
Replaced check with `Path.GetFileName(destinationPath.FullName).Length == 0` to avoid having to reference directory separator characters directly.